modules/__tests__/BrowserHistory-test.js

```diff
@@ -11,187 +12,33 @@ describeHistory('a browser history', () => {
-      it('calls change listeners with the next location', (done) => {
-        TestSequences.GoForward(history, done)
-      })
-    })
+  const cases = [
+    TestSequences.Listen,
+    TestSequences.InitialLocationNoKey,
+    TestSequences.PushNewLocation,
+    TestSequences.PushSamePath,
+    TestSequences.PushState,
+    TestSequences.PushMissingPathname,
+    TestSequences.PushRelativePathname,
+    TestSequences.PushUnicodeLocation,
+    TestSequences.PushEncodedLocation,
+    TestSequences.PushInvalidPathname,
+    TestSequences.ReplaceNewLocation,
+    TestSequences.ReplaceSamePath,
+    TestSequences.ReplaceState,
+    TestSequences.ReplaceInvalidPathname,
+    TestSequences.LocationPathnameAlwaysDecoded,
+    TestSequences.EncodedReservedCharacters,
+    TestSequences.GoBack,
+    TestSequences.GoForward,
+    TestSequences.Before,
+    TestSequences.BeforeGo,
+    TestSequences.BeforeGoBrowser,
+    TestSequences.BeforeHashChangeBrowser,
+  ]
 
-    describe('block', () => {
-      it('blocks all transitions', (done) => {
-        TestSequences.BlockEverything(history, done)
-      })
-    })
-
-    describe('block a POP without listening', () => {
-      it('receives the next location and action as arguments', (done) => {
-        TestSequences.BlockPopWithoutListening(history, done)
-      })
-    })
-  })
-
-  describe('that denies all transitions', () => {
-    const getUserConfirmation = (_, callback) => callback(false)
-
-    let history
-    beforeEach(() => {
-      history = createHistory({
-        getUserConfirmation
-      })
-    })
-
-    describe('clicking on a link (push)', () => {
-      it('does not update the location', (done) => {
-        TestSequences.DenyPush(history, done)
-      })
-    })
-
-    describe('clicking the back button (goBack)', () => {
-      it('does not update the location', (done) => {
-        TestSequences.DenyGoBack(history, done)
-      })
-    })
-
-    describe('clicking the forward button (goForward)', () => {
-      it('does not update the location', (done) => {
-        TestSequences.DenyGoForward(history, done)
-      })
-    })
-  })
-
-  describe('a transition hook', () => {
-    const getUserConfirmation = (_, callback) => callback(true)
-
-    let history
-    beforeEach(() => {
-      history = createHistory({
-        getUserConfirmation
-      })
-    })
-
-    it('receives the next location and action as arguments', (done) => {
-      TestSequences.TransitionHookArgs(history, done)
-    })
-
-    it('cancels the transition when it returns false', (done) => {
-      TestSequences.ReturnFalseTransitionHook(history, done)
-    })
-
-    it('is called when the back button is clicked', (done) => {
-      TestSequences.BackButtonTransitionHook(history, done)
-    })
-
-    it('is called on the hashchange event', (done) => {
-      TestSequences.HashChangeTransitionHook(history, done)
-    })
+  describe('by default', () => {
+    generateTests(cases, createHistory)
```

Comment:
There's nothing "cleaner" about using a loop to generate the test cases. Besides the fact that the code is now less explicit, it's also more difficult to use `describe.only` and `it.only` in development to run only a single test or suite.